### PR TITLE
Add osx_arm64 build instructions

### PR DESCRIPTION
**Pull request by ChristopherMayes**
Following: https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

@conda-forge-admin, please rerender

### CODE REVIEW

**Comment by ChristopherMayes** [comment]
@conda-forge-admin, please rerender

**Comment by bollwyvl** [comment]
You (or I) can update the license to `LGPL-3.0-or-later` to get rid of the linter error

**Comment by ChristopherMayes** [comment]
@bollwyvl I updated the license. But I don't know how to fix the pandoc errors.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ requirements:
     - {{ compiler('cxx') }}
   host:
     - numpy
-    - pypandoc
+    - pypandoc  # [not osx_arm64]
```
Comment (bollwyvl) [inline recipe/meta.yaml]: perhaps `# [not (osx and not x86_64)]`

Comment (ChristopherMayes) [reply recipe/meta.yaml]: Okay that selects it out. But there are still problems -- perhaps I can try more tricks from https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/

Comment (ChristopherMayes) [reply recipe/meta.yaml]: Works!

**Comment by ChristopherMayes** [comment]
@bollwyvl does this look good enough to merge?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,9 +22,15 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - cython                                 # [build_platform != target_platform]
```
Comment (bollwyvl) [inline recipe/meta.yaml]: does it actually use cython and pybind11? these can have knock-on effects if not managed properly. i think if you check that blog post, you'll see the modifier "if needed"...

Comment (ChristopherMayes) [reply recipe/meta.yaml]: @bollwyvl Okay, I removed those and it still works.

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
Thanks for sticking with this one. I'll watch the master build for flake, and hopefully this should be up in a few hours: please open an issue if you run into any further problems.

**Pull request merged**